### DISCUSSION TITLE
Audio Enhancement - Echo and noise cancellation

### DISCUSSION THREAD

**Discussion by DusanGataric** [sipsorcery-org/sipsorcery#452] [Q&A]
Do you plan to support echo and noise cancellation in your SIP project?
I have managed to make a SIP call using your Softphone example but I am experiencing really bad echo issues.

**Comment by sipsorcery** [comment]
No, I don't have any plans to add any audio signal processing to this library. The audio processing path will stop at the encode/decode to raw signed linear samples.

I appreciate it would be useful to have full audio and video processing pipelines but that's a full-time undertaking on its own and I can't even keep up with the network, transport and signalling specifications for real-time communications.

**Reply by DusanGataric** [reply]
Yeah, it's seems like making a complete Voip package, with audio enhancements is immense work.  
I didn't find any SIP solution for C# with adequate echo cancellation yet.
Anyways, thanks for the reply.
